### PR TITLE
Remove JWT authentication method

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
For Cloud, we can only support authentication via OAuth. This PR drops the support for JWT.

As discussed with @aaronrothschild, there are not build-in migration steps. Existing JWT installations will just stop working. An admin has to manually migrate to OAuth.

I did locally test the authentication with OAuth for the per-user case.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-29148

### CODE REVIEW

**Comment by hanzei** [comment]
@larkox The base branch is `release_v1.5.0-cloud` for now. Please let me know if the changes should also go into `master`.

**Comment by DHaussermann** [comment]
@hanzei I found no issues with Fresh Install cases but I'm not sure what's happening with JWT here. If a customer is using JWT it seems this PR just drops it. My impression was that existing JWT config could still be used. 
https://community.mattermost.com/core/pl/wmnhy3ouptbqmpwm6jaxrbhoow

**Review by mickmister — APPROVED** [review]
LGTM :+1:

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,6 @@ module github.com/mattermost/mattermost-plugin-zoom
 go 1.12
 
 require (
-	github.com/dgrijalva/jwt-go v3.2.0+incompatible
```
[the remark this answers is not in the document]
Comment (hanzei) [reply go.mod:6]: `go mod tidy` says no :wink:

**Review by larkox — APPROVED** [review]
LGTM. Thank you very much @hanzei for taking care of this. My original idea was to just remove the settings from the manifest, but this work had to be done eventually.
This will be needed to be merged to master AFTER 1.5 is released.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- JWT removed from the UI
- Previously saved JWT no longer functional
- Regression tested user level oAuth
-  Regression tested OAuth by Account Level App

LGTM!

**Comment by DHaussermann** [comment]
@hanzei unless I'm misunderstanding something, this can merge to `release_v1.5.0-cloud` now. It will merge to `master` once 1.5 is released correct?

**Comment by hanzei** [comment]
Correct @DHaussermann

**Pull request closed without merging**